### ISSUE TITLE
Text-object surround

### ISSUE THREAD

**Issue by musjj** [jeffreytse/zsh-vi-mode#281]
## General information
- Terminal program: tmux 3.4 (tmux-256color)
- Operating system: Linux (6.6.28 <span>#</span>1-NixOS SMP PREEMPT_DYNAMIC Wed Apr 17 09:19:38 UTC 2024 x86_64 GNU/Linux)
- ZSH framework: starship 1.18.2
- ZSH version: zsh 5.9 (x86_64-pc-linux-gnu)
- ZVM version: zsh-vi-mode 0.11.0 

## Basic examination

- [x] I have read through the [README](https://github.com/jeffreytse/zsh-vi-mode) page
- [x] I have the latest version of zsh-vi-mode
- [x] I have tested with another terminal program 

## Problem description

Many implementations of surround plugins in vim/neovim allows you to use surround commands directly with a text object. For example, `ysiw"` will add quotes to the current word. Currently, you need to use visual mode first, like: `viwS"`. It's not any less efficient, but muscle memory is a hell of a drug :)

**Comment by cpakkala** [comment]
Agreed; this is the one reason why I cannot use this plugin yet.  "ysiw" is the most basic surround sequence...

**Comment by jeffreytse** [comment]
Hi @musjj, thanks for your issue about text-object surround, I will try to dive into this ASAP, currently I'm a bit busy.
